charts/node/templates/statefulset.yaml

```
@@ -640,6 +640,22 @@ spec:
             secretName: {{ $fullname }}-{{ .type }}
             defaultMode: 0400
       {{- end }}
+      {{- if .Values.node.chainKeystore.mountInMemory.enabled }}
+        - name: chain-keystore
+          emptyDir:
+            medium: "Memory"
+            {{- if .Values.node.chainKeystore.mountInMemory.sizeLimit }}
+            sizeLimit: {{ .Values.node.chainKeystore.mountInMemory.sizeLimit }}
+            {{- end }}
```

```suggestion
            {{- if and (.Values.node.chainKeystore.mountInMemory.sizeLimit) (semverCompare ">= 1.22-0") }}
            sizeLimit: {{ .Values.node.chainKeystore.mountInMemory.sizeLimit }}
            {{- end }}
```

This uses the capabilities functionality in helm to determine if the minimum version of kubernetes is sufficient to support `sizeLimit`

---

I would advise testing this as I am just doing it from memory - untested.

---

Added a version check when sizeLimit is set